### PR TITLE
Feature/229 ability to like projects

### PR DESCRIPTION
**Pull request by aintabb**
## Description

- Added a new endpoint to the API for ability to like a project
- Fixed some of the typo issues
- Trimmed additional spaces.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

1. Start Postman
2. Create a new request.
3. Select the request type PUT
4. Use "/project/{{projectId}}/like/{{userId}}" as a request template.
5. Run the test
6. You should see the userIds who liked the project in project table.

## Link to issue

Closes: [#229](https://github.com/DigitalExcellence/dex-backend/issues/229)

### CODE REVIEW

**Comment by aintabb** [comment]
@Brend-Smits Please review this pull request and give me some feedback. So that i could continue to progress.

**Review by niraymak — CHANGES_REQUESTED** [review]
Thanks for your first pull request :)

I think you should look into implementing a ManyToMany relationship for project likes.
https://www.learnentityframeworkcore.com/configuration/many-to-many-relationship-configuration

I think there should be a new table called ProjectLike which has references FK_userId, FK_projectId.
All records in this table represent "likes".

After you've done the above, you will be able to get all likes from the project entity.

You can also look into @MeesvanStraten 's and @DaveBouman which is very similar. Their changes are on the branch underneath.
https://github.com/DigitalExcellence/dex-backend/tree/feature/228-follow-users-projects

File: API/Controllers/ProjectController.cs [hunk 1]
```diff
@@ -264,5 +266,95 @@ public async Task<IActionResult> DeleteProject(int projectId)
...
+        /// <summary>
+        /// Like a project with the specified project identifier and
+        /// the user identifier.
+        /// </summary>
+        /// <param name="projectId">The project identifier.</param>
+        /// <param name="userId">The user identifier who likes the project.</param>
+        /// <returns>The project resource result.</returns>
+        [HttpPut("{projectId}/like/{userId}")]
+        [Authorize(Policy = nameof(Defaults.Scopes.ProjectWrite))]
+        public async Task<IActionResult> LikeProject(int projectId, int userId)
```
[the remark this answers is not in the document]
Comment (aintabb) [reply API/Controllers/ProjectController.cs:279]: Thanks a lot for your review @niraymak. I'll review your recommendations and make the necessary corrections.
I appreciate your time and effort for reviewing my pull request.

[the remark this answers is not in the document]
Comment (aintabb) [reply API/Controllers/ProjectController.cs]: I believe I did all the necessary changes. 
I followed @MeesvanStraten 's and @DaveBouman's commits for ability to follow projects and users.

**Comment by Brend-Smits** [comment]
Since #240 is now merged, make sure the documentation is updated!

**Comment by aintabb** [comment]
@Brend-Smits Ok, I will take a look.

**Comment by Brend-Smits** [comment]
@MeesvanStraten since your branch is now merged. It can be used as a blueprint. Could you take lead on reviewing this pull request and provide feedback for @aintabb to process, please?

**Comment by aintabb** [comment]
@niraymak The long wait is finally over. Thanks for the review. @Brend-Smits I could be more useful, if you can speed up the review process.

**Comment by niraymak** [comment]
@aintabb No worries! I created an internal branch called feature/229-ability-to-like-projects on our side. I added new Postman tests and renamed the table / models. Can you merge our branch into your branch? I think it should be ready to review then.

**Comment by aintabb** [comment]
@niraymak Sure, I can do that.

**Comment by aintabb** [comment]
@niraymak OK, I merged the changes from your branch to your branch. It seems you only changed the naming throughout the project. May I ask the reason? When I was designing the new feature, I thought "LikedProjectByUser"  would be more concise and descriptive

**Comment by niraymak** [comment]
> @niraymak OK, I merged the changes from your branch to your branch. It seems you only changed the naming throughout the project. May I ask the reason? When I was designing the new feature, I thought "LikedProjectByUser" would be more concise and descriptive

Thanks for merging! I thought ProjectLike was shorter and describes the same thing. To be honest I think it is obvious a project is liked by a user. Any second opinions on this? @Brend-Smits @waltersajtos 

Thanks for your great code! 😁

**Comment by waltersajtos** [comment]
For now, it is obvious that a project is liked by a user. I can't think of any case that you could like something else on the platform. Although I think that being concise and descriptive is more important than having the shortest name possible. I think it's just a case of personal preference since - as Niray said - it describes the same thing.

**Review by waltersajtos — DISMISSED** [review]

**Comment by aintabb** [comment]
> For now, it is obvious that a project is liked by a user. I can't think of any case that you could like something else on the platform. Although I think that being concise and descriptive is more important than having the shortest name possible. I think it's just a case of personal preference since - as Niray said - it describes the same thing.

Got it! Thanks for the comments @niraymak @waltersajtos I just wanted hear your thinking. That’s all.

**Review by niraymak — COMMENTED** [review]
Only had some issues starting without a database. It crashed the first time, second time it worked. I'm asking the team in Slack if this will cause any issues when merging to develop. 

No other comments. 💯

**Review by MeesvanStraten — APPROVED** [review]

File: API/Resources/UserProjectLikeResource.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        ///    The Project class instance
+        /// </value>
+        public Project LikedProject { get; set; }
+        /// <summary>
+        ///     Gets or sets the user who created the project
+        /// </summary>
+        /// <value>
+        ///    The User class instance
+        /// </value>
+        public User CreatorOfProject { get; set; }
```
[the remark this answers is not in the document]
Comment (aintabb) [reply API/Resources/UserProjectLikeResource.cs:30]: Hi @MeesvanStraten , 
To my way of thinking, the summary explains clearly that "CreatorOfProject" points out the owner of the project.
Let's check the summary together; "Gets or sets the user who **_created the project_**". I think it is clear?

**Review by niraymak — APPROVED** [review]
🥇

**Review by MeesvanStraten — DISMISSED** [review]

**Review by MeesvanStraten — APPROVED** [review]

**Review by niraymak — APPROVED** [review]

**Pull request merged**